Tryouts/Core/Abstractions/ISubscriptionClient.cs

```diff
@@ -9,19 +9,15 @@
 /// 
 /// ********************************************************************************************************
 
-using Google.Protobuf;
-using Subscriptions;
-
 namespace MorganStanley.ComposeUI.Tryouts.Core.Abstractions
 {
     public interface ISubscriptionClient
     {
         Task Connect(string host, int port);
 
-        Task Publish<TMessage>(Topic topic, TMessage msg)
-             where TMessage : IMessage;
+        Task Publish(string topic, string msg);
 
-        IAsyncEnumerable<TMessage> Subscribe<TMessage>(Topic topic)
-            where TMessage : IMessage, new();
+        (IObservable<CommunicationsMessage> messageStream, IDisposable unsubscriber)
```

Comment:
This return type is really confusing. The caller would do
```
(var messageStream, var unsubscriber) = subscriptionClient.Subscribe("...");
```
and then 
```
var unsubscriber2 = messageStream.Subscribe();
```
so would end up with two `IDisposable` unsubscription tokens.

Reply:
I know it is a bit awkward and I'll try to improve it.

Reply:
This still results in awkward code like MessageRouter.Subscribe("myTopic").Subscribe(observer);